sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -1083,6 +1083,21 @@ class DataFrame private[sql](
   }
 
   /**
+   * Returns a new [[DataFrame]] with a column dropped.
+   * This version of drop accepts a Column rather than a name.
+   * This is a no-op if the DataFrame doesn't have a column
+   * with an equivalent expression.
+   * @group dfops
+   * @since 1.4.0
+   */
+  def drop(col: Column): DataFrame = {
+    val colsAfterDrop = schema.filter { field =>
+      !this.col(field.name).expr.semanticEquals(col.expr)
```

Comment:
This is not right. If this data frame is a result of join, then it may have 2 columns with same name, and the `this.col(field.name)` will case a ambiguous-reference exception. You should use `df.logicalPlan.output` instead of `df.schema`.